### PR TITLE
[BENCH-658], [BENCH-642] Call WSM-v2 endpoints to create and delete workspaces

### PR DESCRIPTION
**Pull request by dexamundsen**
[NOTE] - 
Wait till V2 endpoints are deployed in prod before merging this PR

min release version needed - auto-20230605-0530-RC0 (not yet in prod)
Downstream - https://github.com/verily-src/terra-service-ws-manager/tree/auto-20230605-0530
Upstream - https://github.com/DataBiosphere/terra-workspace-manager/tree/31e1606d9e7e6799b13d679e0a5654780ee10019  

-----
- Replace calls to WSM-v1 create workspace + cloud context with v2-create workspace
- Replace calls to WSM-v1 delete workspace with v2-delete workspace
-----

No update needed in Update unit / integration tests

Manual testing - 

AWS
```
terra-cli (dexamundsen/bench-658) >> terra workspace create --platform=AWS --id=dex-ws-aws-1
Workspace successfully created.
ID:                dex-ws-aws-1
…

2023-06-12 13:16:27.358 PDT [main] DEBUG bio.terra.cli.service.utils.HttpUtils - Result: class CreateWorkspaceV2Result {…
       description: Create workspace name: '' id: '1476df8a-cc31-4c49-99c9-af9dc3c01637' and AWS cloud context
        status: RUNNING
…}

2023-06-12 13:17:27.458 PDT [main] DEBUG bio.terra.cli.service.utils.HttpUtils - Request attempt #2
2023-06-12 13:17:27.685 PDT [main] DEBUG bio.terra.cli.service.utils.HttpUtils - Result: class CreateWorkspaceV2Result {…
        description: Create workspace name: '' id: '1476df8a-cc31-4c49-99c9-af9dc3c01637' and AWS cloud context
        status: SUCCEEDED
…}

2023-06-12 13:17:27.988 PDT [main] DEBUG bio.terra.cli.service.utils.HttpUtils - polling with retries completed. jobCompleted = true, timedOut = false
2023-06-12 13:17:27.988 PDT [main] INFO  bio.terra.cli.businessobject.Workspace - Created workspace: .. 

// created 2 resources: notebook & storage

terra-cli (dexamundsen/bench-658) >> terra workspace delete --quiet --workspace=dex-ws-aws-1
ID:                dex-ws-aws-1
…
Workspace successfully deleted.

2023-06-12 14:04:41.447 PDT [main] DEBUG bio.terra.cli.service.utils.HttpUtils - Request attempt #1
2023-06-12 14:04:41.597 PDT [main] DEBUG bio.terra.cli.service.utils.HttpUtils - Result: class JobResult {…
        description: Delete workspace 1476df8a-cc31-4c49-99c9-af9dc3c01637
        status: RUNNING
…}

2023-06-12 14:10:42.627 PDT [main] DEBUG bio.terra.cli.service.utils.HttpUtils - Request attempt #7
2023-06-12 14:10:42.826 PDT [main] DEBUG bio.terra.cli.service.utils.HttpUtils - Result: class JobResult {…
        description: Delete workspace 1476df8a-cc31-4c49-99c9-af9dc3c01637
        status: SUCCEEDED
…}
```

GCP
```
terra-cli (dexamundsen/bench-658) >> terra workspace create --platform=GCP --id=dex-ws-gcp-1
Workspace successfully created.
ID:                dex-ws-gcp-1
…

2023-06-12 13:24:20.723 PDT [main] DEBUG bio.terra.cli.service.utils.HttpUtils - Result: class CreateWorkspaceV2Result {…
        description: Create workspace name: '' id: '94640218-1df6-45c1-b4dc-8d0f474e1e75' and GCP cloud context
        status: RUNNING
…}

2023-06-12 13:25:20.804 PDT [main] DEBUG bio.terra.cli.service.utils.HttpUtils - Request attempt #2
2023-06-12 13:25:20.934 PDT [main] DEBUG bio.terra.cli.service.utils.HttpUtils - Result: class CreateWorkspaceV2Result {…
        description: Create workspace name: '' id: '94640218-1df6-45c1-b4dc-8d0f474e1e75' and GCP cloud context
        status: SUCCEEDED
…}

// created 2 resources: notebook & storage

terra-cli (dexamundsen/bench-658) >> terra workspace delete --quiet —workspace=dex-ws-gcp-1
ID:                dex-ws-gcp-1
…
Workspace successfully deleted.

2023-06-12 13:45:03.967 PDT [main] DEBUG bio.terra.cli.service.utils.HttpUtils - Request attempt #1
2023-06-12 13:45:04.148 PDT [main] DEBUG bio.terra.cli.service.utils.HttpUtils - Result: class JobResult {…
        description: Delete workspace 94640218-1df6-45c1-b4dc-8d0f474e1e75
        status: RUNNING
…}

2023-06-12 13:46:04.302 PDT [main] DEBUG bio.terra.cli.service.utils.HttpUtils - Request attempt #2
2023-06-12 13:46:04.495 PDT [main] DEBUG bio.terra.cli.service.utils.HttpUtils - Result: class JobResult {…
        description: Delete workspace 94640218-1df6-45c1-b4dc-8d0f474e1e75
        status: SUCCEEDED
…}
```

### CODE REVIEW

File: src/main/java/bio/terra/cli/service/WorkspaceManagerService.java [hunk 1]
```diff
@@ -295,14 +275,44 @@ private void logWorkspaceDescription(WorkspaceDescription workspaceWithContext)
...
+                  () -> {
+                    try {
+                      return workspaceApi.getDeleteWorkspaceV2Result(workspaceId, jobId);
+                    } catch (ApiException e) {
+                      throw new RuntimeException(e);
+                    }
+                  },
+                  (result) -> isDone(result.getJobReport()),
+                  WorkspaceManagerService::isRetryable,
+                  // Resource deletion may take 15+ minutes, so poll for uu to 30 minutes.
```
Comment (ddietterich) [inline src/main/java/bio/terra/cli/service/WorkspaceManagerService.java:309]: micronit: uu

**Review by ddietterich — APPROVED** [review]
LGTM

**Review by jmczerk — APPROVED** [review]
Verily `prod` WSM is now at `0.254.758`, so I think this should be good to go.

File: src/main/java/bio/terra/cli/service/WorkspaceManagerService.java (hunk 1 above)
Comment (dexamundsen) [reply src/main/java/bio/terra/cli/service/WorkspaceManagerService.java]: fixed

**Pull request merged**